impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/AbstractSourcedGraphService.java

```diff
@@ -104,4 +234,90 @@ protected void removeRelationship(Constants.Type typeInSource, Relationships.Wel
     protected abstract String getProposedId(Blueprint b);
 
     protected abstract Filter[] initNewEntity(Vertex newEntity, Blueprint blueprint);
+
+    private static class BFSTransitiveClosureSpliterator implements Spliterator<Vertex> {
+        private final Direction direction;
+        private final String[] edgeLabels;
+        Deque<Iterator<Vertex>> pending;
+
+        private BFSTransitiveClosureSpliterator(Iterator<Vertex> pending, Direction direction,
+                                                String[] edgeLabels) {
+            this.direction = direction;
+            this.edgeLabels = edgeLabels;
+            this.pending = new ArrayDeque<>(Arrays.asList(pending));
+        }
+
+        @Override
+        public boolean tryAdvance(Consumer<? super Vertex> action) {
+            Iterator<Vertex> queueToProcess = peekNonEmpty();
+            if (queueToProcess == null) {
+                return false;
+            }
+
+            Vertex toProcess = queueToProcess.next();
+
+            pending.add(toProcess.getVertices(direction, edgeLabels).iterator());
+
+            action.accept(toProcess);
```

Comment:
Also removing the vertices directly in Gremlin should be more efficient than doing it 1by1 with the need of their looking up (global index issue)

st. like
`v1.out('contains').loop(1){true}{true}.remove()`